diff -ruN .gitignore .gitignore
--- .gitignore	1970-01-01 02:00:00.000000000 +0200
+++ .gitignore	2025-08-05 18:13:11.368089637 +0300
@@ -0,0 +1,2 @@
+*
+!.gitignore
\ No newline at end of file
diff -ruN src/contracts/CrossChainForwarder.sol src/contracts/CrossChainForwarder.sol
--- src/contracts/CrossChainForwarder.sol	2025-08-13 14:51:48.742535847 +0300
+++ src/contracts/CrossChainForwarder.sol	2025-08-05 18:13:11.368089637 +0300
@@ -335,36 +335,36 @@
     uint256 destinationChainId,
     uint256 gasLimit,
     ChainIdBridgeConfig[] memory bridgeAdapters
-  ) internal returns (bool) {
+  ) internal virtual returns (bool) {
     bool isForwardedAtLeastOnce = false;
     for (uint256 i = 0; i < bridgeAdapters.length; i++) {
-      (bool success, bytes memory returnData) = bridgeAdapters[i]
-        .currentChainBridgeAdapter
-        .delegatecall(
-          abi.encodeWithSelector(
-            IBaseAdapter.forwardMessage.selector,
-            bridgeAdapters[i].destinationBridgeAdapter,
-            gasLimit,
-            destinationChainId,
-            encodedTransaction
-          )
-        );
+        bool success=true; bytes memory returnData; // (bool success, bytes memory returnData) = bridgeAdapters[i]
+        //.currentChainBridgeAdapter
+        //.delegatecall(
+        //  abi.encodeWithSelector(
+        //    IBaseAdapter.forwardMessage.selector,
+        //    bridgeAdapters[i].destinationBridgeAdapter,
+        //    gasLimit,
+        //    destinationChainId,
+        //    encodedTransaction
+        //  )
+        //);
 
       if (success) {
         isForwardedAtLeastOnce = true;
       } else {
         // it doesn't revert as sending to other bridges might succeed
       }
-      emit TransactionForwardingAttempted(
-        transactionId,
-        envelopeId,
-        encodedTransaction,
-        destinationChainId,
-        bridgeAdapters[i].currentChainBridgeAdapter,
-        bridgeAdapters[i].destinationBridgeAdapter,
-        success,
-        returnData
-      );
+      //emit TransactionForwardingAttempted(
+      //  transactionId,
+      //  envelopeId,
+      //  encodedTransaction,
+      //  destinationChainId,
+      //  bridgeAdapters[i].currentChainBridgeAdapter,
+      //  bridgeAdapters[i].destinationBridgeAdapter,
+      //  success,
+      //  returnData
+      //);
     }
 
     return (isForwardedAtLeastOnce);
@@ -418,11 +418,11 @@
 
       if (!configFound) {
         // preparing fees stream
-        Address.functionDelegateCall(
-          bridgeAdapterConfigInput.currentChainBridgeAdapter,
-          abi.encodeWithSelector(IBaseAdapter.setupPayments.selector),
-          Errors.ADAPTER_PAYMENT_SETUP_FAILED
-        );
+        //Address.functionDelegateCall(
+        //  bridgeAdapterConfigInput.currentChainBridgeAdapter,
+        //  abi.encodeWithSelector(IBaseAdapter.setupPayments.selector),
+        //  Errors.ADAPTER_PAYMENT_SETUP_FAILED
+        //);
 
         bridgeAdapterConfigs.push(
           ChainIdBridgeConfig({
